include/dxc/DXIL/DxilSemantic.h

```diff
@@ -25,24 +25,26 @@ class Semantic {
   using Kind = DXIL::SemanticKind;
 
   enum class CompTy {
-    BoolTy         = 1 << 0,
-    HalfTy         = 1 << 1,
-    Int16Ty        = 1 << 2,
-    FloatTy        = 1 << 3,
-    Int32Ty        = 1 << 4,
-    DoubleTy       = 1 << 5,
-    Int64Ty        = 1 << 6,
-    HalfOrFloatTy  = HalfTy | FloatTy,
-    BoolOrInt32Ty  = BoolTy | Int32Ty,
-    FloatOrInt32Ty = FloatTy | Int32Ty,
-    Int16Or32Ty    = Int16Ty | Int32Ty,
-    AnyIntTy       = BoolTy | Int16Ty | Int32Ty | Int64Ty,
-    AnyFloatTy     = HalfTy | FloatTy | DoubleTy,
-    AnyTy          = AnyIntTy | AnyFloatTy,
+    BoolTy             = 1 << 0,
+    HalfTy             = 1 << 1,
+    Int16Ty            = 1 << 2,
+    FloatTy            = 1 << 3,
+    Int32Ty            = 1 << 4,
+    DoubleTy           = 1 << 5,
+    Int64Ty            = 1 << 6,
+    HalfOrFloatTy      = HalfTy | FloatTy,
+    BoolOrInt32Ty      = BoolTy | Int32Ty,
+    BoolOrInt16Or32Ty  = BoolTy | Int16Ty | Int32Ty,
+    FloatOrInt32Ty     = FloatTy | Int32Ty,
+    Int16Or32Ty        = Int16Ty | Int32Ty,
+    AnyIntTy           = BoolTy | Int16Ty | Int32Ty | Int64Ty,
+    AnyFloatTy         = HalfTy | FloatTy | DoubleTy,
+    AnyTy              = AnyIntTy | AnyFloatTy,
   };
 
   enum class SizeClass {
     Scalar,
+    Vec1,
```

Comment:
Can't we treat Vec1 as Scalar?  I think either one should be legal for all scalar system values, right?  I see some that still only go up to Scalar, so does that mean those won't accept vec1?  Why not?

Also, it would be nice to make the 0 enum value something like Invalid, or the Unknown.  If Scalar aliased Vec1, then vector size would align with the enum values (always easier when looking at raw values, such as in debugger).